R/rdeephaven/R/table_handle_wrapper.R

```diff
@@ -1,104 +1,123 @@
-    #' @return A RecordBatchStreamReader containing the data from the table referenced by this TableHandle.
-    to_arrow_record_batch_stream_reader = function() {
-      ptr <- private$internal_table_handle$get_arrow_array_stream_ptr()
-      rbsr <- RecordBatchStreamReader$import_from_c(ptr)
-      return(rbsr)
-    },
+#' @export
+setMethod(
+  "head",
+  signature = c(x = "TableHandle"),
+  function(x, n, ...) {
+    verify_positive_int("n", n, TRUE)
+    return(new("TableHandle", .internal_rcpp_object = x@.internal_rcpp_object$head(n)))
+  }
+)
 
-    #' @description
-    #' Imports the table referenced by this TableHandle into an Arrow Table.
-    #' @return A Table containing the data from the table referenced by this TableHandle.
-    to_arrow_table = function() {
-      rbsr <- self$to_arrow_record_batch_stream_reader()
-      arrow_tbl <- rbsr$read_table()
-      return(arrow_tbl)
-    },
+#' @export
+setMethod(
+  "tail",
+  signature = c(x = "TableHandle"),
+  function(x, n, ...) {
+    verify_positive_int("n", n, TRUE)
+    return(new("TableHandle", .internal_rcpp_object = x@.internal_rcpp_object$tail(n)))
+  }
+)
 
-    #' @description
-    #' Imports the table referenced by this TableHandle into a dplyr Tibble.
-    #' @return A Tibble containing the data from the table referenced by this TableHandle.
-    to_tibble = function() {
-      rbsr <- self$to_arrow_record_batch_stream_reader()
-      arrow_tbl <- rbsr$read_table()
-      return(as_tibble(arrow_tbl))
-    },
+#' @export
+setMethod(
+  "nrow",
+  signature = c(x = "TableHandle"),
+  function(x) {
+    return(x@.internal_rcpp_object$num_rows())
+  }
+)
 
-    #' @description
-    #' Imports the table referenced by this TableHandle into an R Data Frame.
-    #' @return A Data Frame containing the data from the table referenced by this TableHandle.
-    to_data_frame = function() {
-      arrow_tbl <- self$to_arrow_table()
-      return(as.data.frame(as.data.frame(arrow_tbl))) # TODO: for some reason as.data.frame on arrow table returns a tibble, not a data frame
-    }
-  ),
-  private = list(
-    internal_table_handle = NULL,
-    is_static_field = NULL
-  )
+# TODO: Implement ncol, dim
+
+### TABLEHANDLE CONVERSIONS ###
+
+#' @export
+setMethod(
+  "as_record_batch_reader",
+  signature = c(x = "TableHandle"),
+  function(x, ...) {
+    ptr <- x@.internal_rcpp_object$get_arrow_array_stream_ptr()
+    rbsr <- RecordBatchStreamReader$import_from_c(ptr)
+    return(rbsr)
+  }
+)
+
+#' @export
+setMethod(
+  "as_arrow_table",
+  signature = c(x = "TableHandle"),
+  function(x, ...) {
+    rbsr <- as_record_batch_reader(x)
+    arrow_tbl <- rbsr$read_table()
+    return(arrow_tbl)
+  }
+)
+
+#' @export
+setMethod(
+  "as_tibble",
+  signature = c(x = "TableHandle"),
+  function(x, ...) {
+    rbsr <- as_record_batch_reader(x)
+    arrow_tbl <- rbsr$read_table()
+    return(as_tibble(arrow_tbl))
+  }
+)
+
+#' @export
+setMethod(
+  "as.data.frame",
```

Comment:
for consistency in the API, should we also provide an `as_data_frame` method?

Reply:
My understanding of this is that it's deprecated, seen by `?as_data_frame`. Would not hurt to provide it.

Reply:
I was suggesting adding it for consistency with the other methods we provide.